### PR TITLE
feat/admin/manage-account

### PR DESCRIPTION
**Pull request by hoangtv49**
CRUD for user.
Taiga: https://taiga.aura.network/project/aurascan-2/task/249 
To authorize user add 
@Roles(USER_ROLE.ADMIN) before endpoint

### CODE REVIEW

File: src/auth/role/roles.guard.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    }
+    const { user } = context.switchToHttp().getRequest();
+    const userFound = await this.userService.findOne({
+      where: { email: user?.email },
+    });
+    const userRole = userFound?.role || '';
+    return this.matchRoles(requiredRoles, userRole);
+  }
+
+  matchRoles(roles: string[], userRole: string) {
```
Comment (duonghb53) [inline src/auth/role/roles.guard.ts:26]: Define return value

File: src/auth/role/roles.guard.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { ROLES_KEY, USER_ROLE } from 'src/shared';
+import { UserService } from 'src/components/user/user.service';
```
Comment (duonghb53) [inline src/auth/role/roles.guard.ts:4]: Change absolute path --> relative path.
More: https://github.com/nestjs/typescript-starter/issues/74
Check all case remain

File: src/components/user/controllers/users.controller.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  @ApiOkResponse({
+    description: 'Return user detail.',
+    type: UpdateUserDto,
+  })
+  @ApiNotFoundResponse({
+    description: 'User not found.',
+  })
+  @HttpCode(HttpStatus.OK)
+  async findOne(@Param('id') id: string) {
+    const user = await this.userService.findOne({ where: { id: +id || 0 } });
```
Comment (duonghb53) [inline src/components/user/controllers/users.controller.ts:75]: Move { where ... } to -> service, don't use logic in controller

File: src/auth/role/roles.guard.ts [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-import { UserService } from 'src/components/user/user.service';
+import { ROLES_KEY, USER_ROLE } from '../../shared';
+import { UserService } from '../../components/user/user.service';
 
 @Injectable()
 export class RoleGuard implements CanActivate {
   constructor(private reflector: Reflector, private userService: UserService) {}
 
-  async canActivate(context: ExecutionContext) {
+  async canActivate(context: ExecutionContext): Promise<any> {
```
Comment (duonghb53) [inline src/auth/role/roles.guard.ts:10]: Return `Promise<boolean>` ?

**Review by duonghb53 — APPROVED** [review]
LGTM

**Review by nthduc95 — APPROVED** [review]

**Pull request merged**